packages/http/src/types.ts

```diff
@@ -90,16 +102,37 @@ export class ProblemJsonError extends Error {
     return error;
   }
 
+  public static asValidationIssue(error: Error & { detail: IPrismDiagnostic[]; status?: number }): any {
```

Comment:
The spec for the `problem/json` says that effectively additional field apart from the standard one are accepted; I'd suggest, instead of creating a new function to handle the specific case, we could extend the `fromTemplate` function to add an `additional: unknown` parameter that we can add to the payload, in this way the final code should be kind of simplified.

Reply:
hm.. @XVincentX, did you mean `fromTemplate` or `fromPlainError`?